src/UILayer/QueuedStoreGroup.js

```diff
@@ -0,0 +1,198 @@
+// LICENSE : MIT
+"use strict";
+// polyfill Object.assign
+const ObjectAssign = require("object-assign");
+const assert = require("assert");
+const LRU = require("lru-cache");
+const CHANGE_STORE_GROUP = "CHANGE_STORE_GROUP";
+import Dispatcher from "./../Dispatcher";
+import Store from "./../Store";
+import StoreGroupValidator from "./StoreGroupValidator";
+import {ActionTypes} from "../Context";
+/**
+ * StoreGroup is a **UI** parts of Store.
+ * StoreGroup has event queue system.
+ * It means that StoreGroup thin out change events of stores.
+ * If you want to know all change events, and directly use `store.onChange()`.
+ * @public
+ */
+export default class QueuedStoreGroup extends Dispatcher {
+    /**
+     * Create StoreGroup
+     * @param {Store[]} stores stores are instance of `Store` class
+     */
+    constructor(stores) {
+        super();
+        StoreGroupValidator.validateStores(stores);
+        /**
+         * callable release handlers
+         * @type {Function[]}
+         * @private
+         */
+        this._releaseHandlers = [];
+
+        /**
+         * array of store that emit change in now!
+         * this array is temporary cache in changing the StoreGroup
+         * @type {Store[]}
+         * @private
+         */
+        this._currentChangingStores = [];
+        /**
+         * @type {Store[]}
+         */
+        this.stores = stores;
+        // listen onChange of each store.
+        this.stores.forEach(store => this._registerStore(store));
+        /**
+         * LRU Cache for Store and State
+         * @type {LRU}
+         * @private
+         */
+        this._stateCache = new LRU({
+            max: 100,
+            maxAge: 1000 * 60 * 60
+        });
+        // `this` can catch the events of dispatchers
+        // Because context delegate dispatched events to **this**
+        const didExecutedUseCase = (payload) => {
+            if (payload.type === ActionTypes.ON_DID_EXECUTE_EACH_USECASE) {
+                const parent = payload.parent;
+                // emitChange when root useCase is executed
+                // ignore child useCase is executing
+                if (!parent && this.hasChangingStore) {
```

Comment:
Shall we add `asap` Option?
Currently, we emit change on did executed **root** UseCase.